pkg/mesh-networking/apply/applier.go

```diff
@@ -755,6 +761,234 @@ func getAppliedVirtualMesh(
 	return nil
 }
 
+func getAppliedEastWestIngressGateways(
+	virtualMeshes networkingv1sets.VirtualMeshSet,
+	mesh *discoveryv1.Mesh,
+	destinations discoveryv1sets.DestinationSet,
+) []*discoveryv1.MeshStatus_AppliedIngressGateway {
+	// Check that this mesh belongs to a virtual mesh
+	if mesh.Status.GetAppliedVirtualMesh() == nil {
+		return nil
+	}
+	virtualMesh, err := virtualMeshes.Find(mesh.Status.GetAppliedVirtualMesh().GetRef())
+	if err != nil {
+		// should never happen
+		return nil
+	}
+
+	var selectedIngressGatewayList []*discoveryv1.MeshStatus_AppliedIngressGateway
+	// Resolve all of the VM’s ingress gateway selectors to a list of (Destination, tls port) tuples that belong to the mesh
```

Comment:
Super Nit, I was very confused why we were talking about Virtual Machines 😆 
```suggestion
	// Resolve all of the VirtualMesh’s ingress gateway selectors to a list of (Destination, tls port) tuples that belong to the mesh
```